openassessment/data.py

```diff
@@ -333,4 +337,135 @@ def _use_read_replica(self, queryset):
             queryset.using("read_replica")
             if "read_replica" in settings.DATABASES
             else queryset
-        )
\ No newline at end of file
+        )
+
+
+class AggregateOraData(object):
+    """
+    Aggregate all the ORA data into a single table.
+
+    Only works on MySQL databases.
+    """
+
+    @classmethod
+    def _read_replica_cursor_if_available(cls):
+        """
+        If there's a read replica that can be used, return a cursor to that.
+        Otherwise, return a cursor to the regular database.
+        """
+        try:
+            cursor = db.connections['read_replica'].cursor()
+        except ConnectionDoesNotExist:
+            cursor = db.connection.cursor()
+        return cursor
+
+    @classmethod
+    def collect_ora2_data(cls, course_id):
+        """
+        Query MySQL database for aggregated ora2 response data.
+        """
+        cursor = cls._read_replica_cursor_if_available()
+
+        # Syntax unsupported by other vendors such as SQLite test db
+        if db.connection.vendor != 'mysql':
+            return '', ['']
+
+        # The raw query has two statements separated by a semi-colon.
+        # We would like to run them separately through the Django ORM.
+        raw_queries = cls.RAW_QUERY.split(';')
```

Comment:
Any reason not to split RAW_QUERY into 2 variables with a reasonable name for each?